x-pack/plugins/security_solution/cypress/tasks/create_new_rule.ts

```diff
@@ -282,19 +289,132 @@ export const fillDefineEqlRuleAndContinue = (rule: CustomRule) => {
   cy.get(EQL_QUERY_INPUT).should('not.exist');
 };
 
+/**
+ * Fills in the indicator match rows for tests by giving it an optional rowNumber,
+ * a indexField, a indicatorIndexField, and an optional validRows which indicates
+ * which row is valid or not.
+ *
+ * There are special tricks below with Eui combo box:
+ * cy.get(`button[title="${indexField}"]`)
+ * .should('be.visible')
+ * .then(([e]) => e.click());
+ *
+ * To first ensure the button is there before clicking on the button. There are
+ * race conditions where if the Eui drop down button from the combo box is not
+ * visible then the click handler is not there either, and when we click on it
+ * that will cause the item to _not_ be selected. Using a {enter} with the combo
+ * box also does not select things from EuiCombo boxes either, so I have to click
+ * the actual contents of the EuiCombo box to select things.
+ */
+export const fillIndicatorMatchRow = ({
+  rowNumber,
+  indexField,
+  indicatorIndexField,
+  validColumns,
+}: {
+  rowNumber?: number; // default is 1
+  indexField: string;
+  indicatorIndexField: string;
+  validColumns?: 'indexField' | 'indicatorField' | 'both' | 'none'; // default is both are valid entries
+}) => {
+  const computedRowNumber = rowNumber == null ? 1 : rowNumber;
+  const computedValueRows = validColumns == null ? 'both' : validColumns;
```

Comment:
You can't use `??` within Cypress from what I saw. It would blow up about those types. I don't know where the version of TypeScript or its features are configured within Cypress but I don't see `??` anywhere in there and when I tried to use it, it would 💥 blow up.

I would like to upgrade and use these though.